### DISCUSSION TITLE
Upload to printer SD in V3.0

### DISCUSSION THREAD

**Discussion by RoP09** [luc-github/ESP3D#689] [Questions]
I'd like to upload files to the printers SD through ESP3D even if it is slow, but somehow the menu item for upload is not visible once I switch to printer SD. I can start prints or delete files, but upload is not possible.

Has this to be enabled in the configuration.h file somehow? I was looking there but could not find any suitable define 
(this is on ESP3D 3.0 and Marlin/Jyers FW on a Creality Ender 3 V2 4.2.2)

**Comment by Legsmaniac** [comment]
I would honestly recommend you scrap that idea because, yes, it IS SOOOOOOOO slow.

Get a FlashAir WiFi SD Card instead and an microSD to SD adapter.  Transfer your files from anywhere wirelessly to your SD card directly. 
Have a look here - https://blog.prusaprinters.org/easy-wireless-printing-with-flashair-sd-cards_8603/
It's what I do.  So much faster than ESP3D

**Comment by luc-github** [comment]
@RoP09 I have updated the status of V3 :  https://github.com/luc-github/ESP3D/discussions/540#discussioncomment-1604767

```
UPLOAD_SERIAL_SD (implemented in 2.1 but not available yet in 3.0)
the access to SD is done using M28/M29 protocol
e.g: Default 3D printer protocol

It is too slow and ask a lot work depending FW, also now this upload is almost always corrupted when using Serial TFT so it is pretty useless for most of people,  so I won't implemented it.

I prefer to spend dev time on  https://github.com/luc-github/ESP3D/issues/575 instead, to have real benefit, and no more data corruption due to external queries
```

**Reply by RoP09** [reply]
understand ,thank you<!
